pyodide-build/pyodide_build/xbuildenv.py

```diff
@@ -0,0 +1,332 @@
+import json
+import shutil
+import subprocess
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+import requests
+from pyodide_lock import PyodideLockSpec
+
+from . import build_env
+from .create_pypi_index import create_pypi_index
+from .logger import logger
+
+XBUILDENV_URL = "https://github.com/pyodide/pyodide/releases/download/{version}/xbuildenv-{version}.tar.bz2"
+CDN_BASE = "https://cdn.jsdelivr.net/pyodide/v{version}/full/"
+
+
+class CrossBuildEnvManager:
+    """
+    Manager for the cross-build environment.
+    """
+
+    def __init__(self, env_dir: str | Path) -> None:
+        self.env_dir = Path(env_dir).resolve()
+
+        try:
+            self.env_dir.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            raise ValueError(
+                f"Failed to create cross-build environment at {self.env_dir}"
+            ) from e
+
+    @property
+    def symlink_dir(self):
+        """
+        Returns the path to the symlink that points to the currently used xbuildenv version.
+        """
+        return self.env_dir / "xbuildenv"
+
+    @property
+    def current_version(self) -> str | None:
+        """
+        Returns the currently used xbuildenv version.
+        """
+        if not self.symlink_dir.exists():
+            return None
+
+        return self.symlink_dir.resolve().name
+
+    def _path_for_version(self, version: str) -> Path:
+        """Returns the path to the xbuildenv for the given version."""
+        return self.env_dir / version
+
+    def list_versions(self) -> list[str]:
+        """
+        List the downloaded xbuildenv versions.
+
+        TODO: add a parameter to list only compatible versions
+        """
+        versions = []
+        for version_dir in self.env_dir.glob("*"):
+            if not version_dir.is_dir() or str(version_dir) == str(self.symlink_dir):
```

Comment:
You can just compare the paths directly:
```suggestion
            if not version_dir.is_dir() or version_dir == self.symlink_dir:
```